bundles/binding/org.openhab.binding.isy/src/main/java/org/openhab/binding/isy/internal/ErrorHandler.java

```diff
@@ -0,0 +1,151 @@
+/**
+ * Copyright (c) 2010-2016 by the respective copyright holders.
+ *
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+package org.openhab.binding.isy.internal;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.universaldevices.common.UDClientStatus;
+import com.universaldevices.resources.errormessages.ErrorEventListener;
+import com.universaldevices.resources.errormessages.Errors;
+import com.universaldevices.upnp.UDControlPoint;
+import com.universaldevices.upnp.UDProxyDevice;
+
+/**
+ * Error handler for the ISY client
+ * 
+ * @author Tim Diekmann
+ * @since 1.9.0
+ */
+public class ErrorHandler implements ErrorEventListener {
+
+    private final Logger logger = LoggerFactory.getLogger(getClass());
+
+    private int socket_open_failed_count = 0;
+
+    private void offLine(final int status, Object device) {
+        String err = Errors.getErrorMessage(status);
+
+        this.logger.warn("Warning: " + status + (err == null ? " " : err));
+
+        if (UDClientStatus.isBusy()) {
+            return;
+        }
+        if (status == 1050) {
+            this.socket_open_failed_count++;
+        }
+        if (this.socket_open_failed_count < 3) {
+            return;
+        }
+        this.socket_open_failed_count = 0;
+
+        if (device == null) {
+            device = UDControlPoint.firstDevice;
+        }
+        if (device != null && device instanceof UDProxyDevice) {
+            UDControlPoint.getInstance().offLine((UDProxyDevice) device);
+        }
+
+    }
+
+    private void humanInterventionRequired(final int status, final String msg) {
+        String err = Errors.getErrorMessage(status);
+        this.logger.error("Human Intervention Required: " + status + (err == null ? " " : err));
+        if (msg != null) {
+            this.logger.error(msg);
+        }
+    }
+
+    private void warning(final int status, final String msg) {
+        String err = Errors.getErrorMessage(status);
+
+        this.logger.warn("Warning: " + status + (err == null ? " " : err));
+        if (msg != null) {
+            this.logger.warn(msg);
+        }
+    }
+
+    private void fatalError(final int status, final String msg) {
+        String err = Errors.getErrorMessage(status);
+        this.logger.error("Fatal Error: " + status + (err == null ? " " : err));
```

Comment:
I'm not sure what you mean by parameterized logging?  Do the logger apis accept varying number of parameters?